source/topic6.rst

```diff
@@ -14,10 +14,33 @@ Implementing a Stack --- Array Container
 ========================================
 
 
+Implementation
+--------------
+
+* Let's start our implementation of the ``ArrayStack``
+
+.. code-block:: Java
+    :linenos:
+    :emphasize-lines: 1
+
+    public class ArrayStack<T> implements Stack<T> {
+
+        private static final int DEFAULT_CAPACITY = 100;
+        private T[] stack;
+        private int top;
+
+    }
+
+
+* We want our ``ArrayStack`` to implement the ``Stack`` interface
+    * This way anyone using our implementation knows that it is a ``Stack``
```

Comment:
Maybe not the right time for the course, but this is the first time I see `extends`, so I comment:

_is a_ is a pretty big deal here. This is the **L** from [SOLID](https://en.wikipedia.org/wiki/SOLID). `ArrayStack` **is a** `Stack`, meaning anything that expects a `Stack` can get an `ArrayStack` and be happy without knowing what implementation it is -- _it is a `Stack`_.

Reply:
Will defo add.